client/src/components/RankingPage.js

```diff
@@ -1,10 +1,150 @@
 import '../styles/App.css';
-import React from 'react';
+import '../styles/HomePage.css';
+import '../styles/RankingPage.css';
+import React, { useState, useEffect } from 'react';
+import { DragDropContext } from '@hello-pangea/dnd';
+import recipeService from '../services/recipes';
+import categoryService from '../services/categories';
+import RankingList from './ranking/RankingList';
+import { ReactComponent as RankingGraphic } from '../graphics/ranking_graphic.svg';
+import ToggleEdit from './ranking/ToggleEdit';
+import RankingInfo from './ranking/RankingInfo';
 
 const RankingPage = () => {
+
+  const [categories, setCategories] = useState([]);
+  const [recipes, setRecipes] = useState([]);
+  const [filteredRecipes, setFilteredRecipes] = useState({ranked: [], unranked: []});
+  const [activeCategory, setActiveCategory] = useState(null);
+  const [isDragDisabled, setIsDragDisabled] = useState(true);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const recipes = await recipeService.getAll();
+        setRecipes(recipes);
+        const categories = await categoryService.getAll();
+        setCategories(categories);
+        setActiveCategory(categories.find(cat => cat.name === 'All'));
+      } catch (error) { // Problem connecting to the server
+        console.log(error);
+      }
+    })();
+  }, []);
+
+
+  /* Filter recipes based on category */
+  useEffect(() => {
+    /* don't run the effect before data has been fetched from backend */
+    if (!activeCategory) {
+      return;
+    }
+
+    const filtered = recipes.filter(r => r.categories.map(cat => cat.name).includes(activeCategory.name));
+    setFilteredRecipes({
+      ranked: activeCategory.rankedRecipes.map(rId => filtered.find(r => r.id === rId)),
+      unranked: filtered.filter(r => activeCategory.unrankedRecipes.includes(r.id))
+    });
+  }, [activeCategory]);
+  
+
+  const createCategoryCard = (category) => {
```

Comment:
Since this component is almost identical to the one in HomePage, could it be moved to its own file? The `setActiveCategory` or `setActiveCategories` functions could be provided as props.

Reply:
This component uses a single active category as props while the component in home pages uses a list. Thus, it might be too complicated to combine these to a single component.